File: src/main/java/com/terraforming/ares/model/action/ActionInputData.java
package com.terraforming.ares.model.action;

import lombok.Builder;
import lombok.Data;

/**
 * Created by oleksii.nikitin
 * Creation date 15.05.2022
 */
@Data
@Builder
public class ActionInputData {
    private ActionInputDataType type;
    private int min;
    private int max;
}

File: frontend/src/app/data/ActionInputData.ts
import {ActionInputDataType} from './ActionInputDataType';

export class ActionInputData {
  constructor(type: ActionInputDataType, min: number, max: number) {
    this.type = type;
    this.min = min;
    this.max = max;
  }

  type: ActionInputDataType;
  min: number;
  max: number;
}
